scriptmodules/supplementary/sdl2.sh

```diff
@@ -31,18 +31,29 @@ function get_arch_sdl2() {
     echo "$(dpkg --print-architecture)"
 }
 
-function depends_sdl2() {
+function _list_depends_sdl2() {
     # Dependencies from the debian package control + additional dependencies for the pi (some are excluded like dpkg-dev as they are
     # already covered by the build-essential package retropie relies on.
-    local depends=(devscripts debhelper dh-autoreconf libasound2-dev libudev-dev libibus-1.0-dev libdbus-1-dev fcitx-libs-dev)
+    local depends=(libasound2-dev libudev-dev libibus-1.0-dev libdbus-1-dev fcitx-libs-dev libsndio-dev)
     # these were removed by a PR for vero4k support (cannot test). Needed though at least for for RPI and X11
-    ! isPlatform "vero4k" && depends+=(libx11-dev libxcursor-dev libxext-dev libxi-dev libxinerama-dev libxrandr-dev libxss-dev libxt-dev libxxf86vm-dev libgl1-mesa-dev)
-    isPlatform "rpi" && depends+=(libegl1-mesa-dev libgles2-mesa-dev libglu1-mesa-dev libraspberrypi-dev)
-    isPlatform "mali" && depends+=(mali-fbdev)
+    ! isPlatform "vero4k" && depends+=(libx11-dev libxcursor-dev libxext-dev libxi-dev libxinerama-dev libxkbcommon-dev libxrandr-dev libxss-dev libxt-dev libxv-dev libxxf86vm-dev libgl1-mesa-dev)
+    isPlatform "gles" || isPlatform "gl" && depends+=(libegl1-mesa-dev libgles2-mesa-dev)
+    isPlatform "gl" || isPlatform "rpi" && depends+=(libgl1-mesa-dev libglu1-mesa-dev)
     isPlatform "kms" || isPlatform "rpi" && depends+=(libdrm-dev libgbm-dev)
-    isPlatform "x11" && depends+=(libpulse-dev libegl1-mesa-dev libgles2-mesa-dev libglu1-mesa-dev)
+    isPlatform "x11" && depends+=(libpulse-dev libwayland-dev)
+
+    echo "${depends[@]}"
+}
+
+function depends_sdl2() {
+    # install additional packages that are needed, but may be unsuitable as debian package dependencies due to distribution oddities
+    local depends=(devscripts debhelper dh-autoreconf)
+
+    isPlatform "mali" && depends+=(mali-fbdev)
+    isPlatform "rpi" && depends+=(libraspberrypi-dev)
     isPlatform "vero4k" && depends+=(vero3-userland-dev-osmc)
-    getDepends "${depends[@]}"
+
+    getDepends "$(_list_depends_sdl2) ${depends[@]}"
```

Comment:
Each parameter expansion should be quoted independently, like this:
```suggestion
    getDepends "$(_list_depends_sdl2)" "${depends[@]}"
```

Reply:
Actually, `$(_list_depends_sdl2)` returns a space-separated list, right?
It should not be quoted then.

Reply:
Yes (to the latter). I've updated the PR to ensure that `$(_list_depends_sdl2)` is not quoted, thanks.